src/client.ts

```diff
@@ -463,6 +475,225 @@ export class McpdClient {
+   * 2. Identify which server each template belongs to
+   *
+   * This method automatically filters out unhealthy servers by checking their health
+   * status before fetching templates. Unhealthy servers are silently skipped to ensure
+   * the method returns quickly without waiting for timeouts on failed servers.
+   *
+   * Servers that don't implement resource templates (return 501 Not Implemented) are
+   * silently skipped, allowing this method to work with mixed server types.
+   *
+   * Template fetches from multiple servers are executed concurrently for optimal performance.
+   *
+   * @param options - Optional configuration
+   * @param options.servers - Array of server names to include. If not specified, includes all servers.
+   * @returns Array of resource template schemas with transformed names (serverName__templateName). Only includes templates from healthy servers that support them.
+   * @throws {ConnectionError} If unable to connect to the mcpd daemon
+   * @throws {TimeoutError} If requests to the daemon time out
+   * @throws {AuthenticationError} If API key authentication fails
+   * @throws {McpdError} If health check or initial server listing fails
+   *
+   * @example
+   * ```typescript
+   * // Get all resource templates from all servers
+   * const allTemplates = await client.getResourceTemplates();
+   * // Returns: [
+   * //   { name: "github__file", uriTemplate: "file:///{path}", _serverName: "github", ... }
+   * // ]
+   *
+   * // Get templates from specific servers only
+   * const someTemplates = await client.getResourceTemplates({ servers: ['github'] });
+   *
+   * // Original template name "file" becomes "github__file"
+   * // This prevents clashes if multiple servers have templates with the same name
+   * ```
+   */
+  async getResourceTemplates(options?: {
+    servers?: string[];
+  }): Promise<ResourceTemplate[]> {
+    const { servers } = options || {};
+
+    // Get healthy servers (fetches list if not provided, then filters by health).
+    const healthyServers = await this.#getHealthyServers(servers);
+
+    // Fetch resource templates from all healthy servers in parallel.
+    const results = await Promise.allSettled(
+      healthyServers.map(async (serverName) => ({
+        serverName,
+        templates: await this.#getResourceTemplatesByServer(serverName),
+      })),
+    );
+
+    // Process results and transform template names.
+    const allTemplates: ResourceTemplate[] = results.flatMap((result) => {
+      if (result.status === "fulfilled") {
+        const { serverName, templates } = result.value;
+        return templates.map((template) => ({
+          ...template,
+          name: `${serverName}__${template.name}`,
+          _serverName: serverName,
+          _templateName: template.name,
+        }));
+      } else {
+        console.warn(
+          `Failed to get resource templates for server:`,
+          result.reason,
+        );
+        return [];
+      }
+    });
```

Comment:
nit: just a suggestion for another pattern we could use to avoid the if/else when using `Promise.allSettled` is to use `.filter`


```suggestion
const allTemplates: ResourceTemplate[] = results
  .filter((result) => result.status === "fulfilled")
  .flatMap((result) => {
    const { serverName, templates } = result.value;
    return templates.map((template) => ({
      ...template,
      name: `${serverName}__${template.name}`,
      _serverName: serverName,
      _templateName: template.name,
    }));
  });

results
  .filter((result) => result.status === "rejected")
  .forEach((result) => {
    console.warn(`Failed to get resource templates for server:`, result.reason);
  });

```